.github/workflows/linter.yml

```diff
@@ -0,0 +1,19 @@
+name: golangci-lint
+on:
+    push:
+      branches: [ "main" ]
+    pull_request:
+      branches: [ "main" ]
```

Comment:
Sometimes I'm opening a pull request on top of some other branch, I think we lose nothing if we handle this case too.

Reply:
Agree that early warnings are good if it's a cheap cost to run them :-)